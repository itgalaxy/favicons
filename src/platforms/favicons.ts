import { FaviconHtmlElement } from "../index";
import { FaviconOptions, NamedIconOptions } from "../config/defaults";
import { transparentIcon, transparentIcons } from "../config/icons";
import { OptionalMixin, Platform, uniformIconOptions } from "./base";

const ICONS_OPTIONS: (NamedIconOptions & OptionalMixin)[] = [
  { name: "favicon.ico", ...transparentIcons(16, 24, 32, 48, 64) },
  { name: "favicon-16x16.png", ...transparentIcon(16) },
  { name: "favicon-32x32.png", ...transparentIcon(32) },
  { name: "favicon-48x48.png", ...transparentIcon(48) },
  { name: "favicon.svg", ...transparentIcon(1024), optional: true }, // arbitrary size. if more than one svg source is given, the closest to this size will be picked.
];

export class FaviconsPlatform extends Platform {
  constructor(options: FaviconOptions) {
    super(
      options,
      uniformIconOptions(options, options.icons.favicons, ICONS_OPTIONS),
    );
  }

  override async createHtml(): Promise<FaviconHtmlElement[]> {
    return this.iconOptions.map(({ name, ...options }) => {
      if (name.endsWith(".ico")) {
        // prettier-ignore
        return `<link rel="icon" type="image/x-icon" href="${this.cacheBusting(this.relative(name))}">`;
      } else if (name.endsWith(".svg")) {
        // prettier-ignore
        return `<link rel="icon" type="image/svg+xml" href="${this.cacheBusting(this.relative(name))}">`;
      }

      const { width, height } = options.sizes[0];

      // prettier-ignore
      return `<link rel="icon" type="image/png" sizes="${width}x${height}" href="${this.cacheBusting(this.relative(name))}">`;
    });
  }
}
