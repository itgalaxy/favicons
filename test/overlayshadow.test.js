import favicons from "../src";
import { logo_png } from "./util";

test("should allow configuring 'overlayShadow'", async () => {
  expect.assertions(1);

  const result = await favicons(logo_png, {
    icons: {
      android: {
        background: true,
        mask: true,
        overlayShadow: true,
      },
      appleIcon: false,
      appleStartup: false,
      coast: false,
      firefox: false,
      windows: false,
      favicons: false,
      yandex: false,
    },
  });

  await expect(result).toMatchFaviconsSnapshot();
});
