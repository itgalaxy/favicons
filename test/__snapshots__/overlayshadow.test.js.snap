// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow configuring 'overlayShadow' 10`] = `
Array [
  Object {
    "contents": null,
    "name": "android-chrome-36x36.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-48x48.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-72x72.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-96x96.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-144x144.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-192x192.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-256x256.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-384x384.png",
  },
  Object {
    "contents": null,
    "name": "android-chrome-512x512.png",
  },
]
`;
